driver/main.c

```diff
@@ -1796,6 +1804,11 @@ static int record_event_consumer(struct ppm_consumer_t *consumer,
 	int32_t cbres = PPM_SUCCESS;
 	int cpu;
 
+	if (tp_type < TP_VAL_INTERNAL && !(consumer->tracepoints_attached & (1 << tp_type)))
```

Comment:
Check if the consumer is attached to the tracepoint, or if the tracepoint is a canary value.